Feat/allow renaming project path

pranavtartey opened the pull request:
## Description

Closes: #1425 

1. Added the popup while the user is trying to change the path from the settings.
2. Shows popup to confirm path change.
3.  Copies the folder and the content inside to the updated address while keeping the original copy.
4. After the confirmation shows toast message for the path update.
5. Popup and toast are supported in all four languages.

## Related Issues

<!-- Link any related issues using GitHub keywords (e.g., "closes #123", "fixes #456", "related to #789") -->

## Type of Change

<!-- Put an `x` in the boxes that apply -->

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Release
- [ ] Refactor
- [ ] Other (please describe):

## Testing

<!-- Describe the tests you ran or the steps to verify your changes -->

## Screenshots (if applicable)

<!-- Add screenshots to help explain your changes -->

![flip-buttons-screenshot](https://github.com/user-attachments/assets/37ede9a0-9d92-4d44-a01e-7c28f6f5e309)

![after-path-update](https://github.com/user-attachments/assets/c54aa47b-6de5-423a-b119-a9611da64fec)


## Additional Notes

<!-- Add any other context about the PR here -->

<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Adds feature to rename project paths with confirmation dialog, toast notifications, and localization support.
> 
>   - **Behavior**:
>     - Adds `moveFolder()` in `index.ts` to copy project folder to a new path.
>     - Implements `MOVE_PROJECT_FOLDER` channel in `code.ts` to handle folder move requests.
>     - Displays confirmation dialog and toast notification in `Project/index.tsx` for path updates.
>   - **UI Components**:
>     - Adds `AlertDialog` for path update confirmation in `Project/index.tsx`.
>     - Uses `toast` for success/error notifications in `Project/index.tsx`.
>   - **Localization**:
>     - Updates translation files (`en`, `ja`, `ko`, `zh`) to support new dialog and toast messages.
>   - **Misc**:
>     - Adds `MOVE_PROJECT_FOLDER` to `MainChannels` in `ipc.ts`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=onlook-dev%2Fonlook&utm_source=github&utm_medium=referral)<sup> for 43e2c44fd74abaec8cf86108c5c51c792c48d153. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

spartan-uyen commented:
**Title**: Confirmation does not disappear after selecting new project path

**Summary**: After selecting a new path for the project, the confirmation message stays on screen and remains clickable, creating confusion for the user about whether the action was completed.

**Steps to Reproduce:**
1. Open the project settings.
2. Select a new path for the project.
3. Click the Confirm button
4. Observe the confirmation message

**Actual Result:**
The confirmation message does not disappear.
It remains clickable, which may lead the user to believe the action hasn't been completed.

**Expected Result:**
The confirmation should disappear automatically after a successful update.
A success message should be displayed briefly to notify the user that the change was applied.

spartan-uyen commented:
Hi, I have a concern:
In the case where I change the path of a project and then delete the corresponding folder, will there be an error or some kind of notification to inform the user when they try to use the project in our application?
Currently, if I delete the folder and then try to change the path again, I receive an error: "Failed to update path." and can't update the path afterward.
Do you have any suggestions on how to handle this scenario?

**Steps to reproduce:**
1. Create a new project
2. Change the path of the project
3. Go to the file explorer and delete the folder at the new path
4. Reopen the project in the application
5. Try to change the path to a different (valid) folder

pranavtarteyroxiler commented:
> **Title**: Confirmation does not disappear after selecting new project path
> 
> **Summary**: After selecting a new path for the project, the confirmation message stays on screen and remains clickable, creating confusion for the user about whether the action was completed.
> 
> **Steps to Reproduce:**
> 
> 1. Open the project settings.
> 2. Select a new path for the project.
> 3. Click the Confirm button
> 4. Observe the confirmation message
> 
> **Actual Result:** The confirmation message does not disappear. It remains clickable, which may lead the user to believe the action hasn't been completed.
> 
> **Expected Result:** The confirmation should disappear automatically after a successful update. A success message should be displayed briefly to notify the user that the change was applied.

**My Fix:**
I am trying to fix these type of issues in copy process, by creating a copy manager (somewhat same as CreateManager) that will show the loading when the user is confirming the path change and will give a good success message on success or otherwise. So I think the clickable modal will no longer be a problem.

**Need Suggestions:**
Is there anything else that you would suggest that might work better.

Kitenite reviewed (commented):
Hey @pranavtartey , this looks really good. Just 2 last notes.

1. Could you refactor the Moving folder component into its own file, looks like the functionality got pretty large. 
2. When selecting a new folder, there is a situation we want to handle which is that the folder has been moved in the filesystem already and we just want to point to that new location. Could we make the move/copy folder optional by making it a checkbox in the confirmation? Similar pattern to when deleting the project you can select whether the folder gets deleted as well. Does this make sense?

<img width="240" alt="image" src="https://github.com/user-attachments/assets/611ca5a6-9c31-43ed-8e0d-628ca0527378" />

<img width="786" alt="image" src="https://github.com/user-attachments/assets/7180b8f6-b819-4c7e-95a2-10145074c4c6" />

pranavtartey commented:
> Hey @pranavtartey , this looks really good. Just 2 last notes.
> 
> 1. Could you refactor the Moving folder component into its own file, looks like the functionality got pretty large.
> 2. When selecting a new folder, there is a situation we want to handle which is that the folder has been moved in the filesystem already and we just want to point to that new location. Could we make the move/copy folder optional by making it a checkbox in the confirmation? Similar pattern to when deleting the project you can select whether the folder gets deleted as well. Does this make sense?
> 
> <img alt="image" width="240" src="https://private-user-images.githubusercontent.com/31864905/435434299-611ca5a6-9c31-43ed-8e0d-628ca0527378.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUxNTg2MDIsIm5iZiI6MTc0NTE1ODMwMiwicGF0aCI6Ii8zMTg2NDkwNS80MzU0MzQyOTktNjExY2E1YTYtOWMzMS00M2VkLThlMGQtNjI4Y2EwNTI3Mzc4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIwVDE0MTE0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3YTM0YmMzMGExYjE4MjdmN2I2YzcwZTM4ZGY4MWI5N2YzNmUyOWY0ZWYwNjRkNjFkOTQzODc2NzRhMzk4MmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tHDGcAu2wfBVuEI70V4Zi6Gl1Tv7Bel_jT8fHNdVT5s"> <img alt="image" width="786" src="https://private-user-images.githubusercontent.com/31864905/435434328-7180b8f6-b819-4c7e-95a2-10145074c4c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUxNTg2MDIsIm5iZiI6MTc0NTE1ODMwMiwicGF0aCI6Ii8zMTg2NDkwNS80MzU0MzQzMjgtNzE4MGI4ZjYtYjgxOS00YzdlLTk1YTItMTAxNDUwNzRjNGM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIwVDE0MTE0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmZDg3YzJmNDEwMjEzMDQ3MDZlMzI3M2RmNDgwMmY0MTdmZmUwNjQyZThjNzg3NmM2YTA0MDA2MzQ3MjFjODImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.GdTkTyUqC90j0EwX0TInZCV6duQHTWsVIVWKaD_qJK8">

_Hey @Kitenite, Thanks for the suggestions! The changes are in. Please take a final look_

**Changes:**

1. Added a Copy component in the project folder that holds all the logic for the copy/move project.
2. Added a checkbox (making the copy process optional). Now you can either make a new copy of the project and point to the updated address/location or you can uncheck the checkbox and simply point to the new address without actually creating a copy.

**Reference Images:**


![running instance copy checkbox](https://github.com/user-attachments/assets/5f49d9c1-0013-44d4-ad72-f1fd2badb12f)

![optional copy checkbox instance not running](https://github.com/user-attachments/assets/1181084b-cbd1-4d49-817f-afd772a1c00f)

![ja-copy checkbox instance running](https://github.com/user-attachments/assets/efee4b73-ae85-4a6a-a724-1eeb712adc79)

Kitenite commented:
Hello, I am migrating the desktop app to a new repository. This repository will now focus on the web version. This PR has been migrated to the new repository: https://github.com/onlook-dev/desktop/pull/15. Sorry for the inconvenience, rest assure your work here will also be adapted for the web version (if it hasn't already).
